core/trino-main/src/main/java/io/trino/operator/aggregation/AggregationMaskCompiler.java

```diff
@@ -178,7 +178,7 @@ public static Constructor<? extends AggregationMaskBuilder> generateAggregationM
         body.append(new IfStatement()
                 .condition(maskBlock.instanceOf(DictionaryBlock.class))
                         .ifTrue(new BytecodeBlock()
-                                .append(maskValueBlock.set(maskBlock.cast(DictionaryBlock.class).invoke("getDictionary", ValueBlock.class, position).cast(ByteArrayBlock.class)))
+                                .append(maskValueBlock.set(maskBlock.cast(DictionaryBlock.class).invoke("getDictionary", ValueBlock.class).cast(ByteArrayBlock.class)))
```

Comment:
this is not sufficient fix.

Reply:
see https://github.com/trinodb/trino/pull/21049, it passes https://github.com/trinodb/trino/issues/21002#issuecomment-1994003746